packet.go

```diff
@@ -820,10 +827,11 @@ type sshFxpDataPacket struct {
 }
 
 func (p sshFxpDataPacket) MarshalBinary() ([]byte, error) {
-	b := []byte{sshFxpData}
-	b = marshalUint32(b, p.ID)
-	b = marshalUint32(b, p.Length)
-	b = append(b, p.Data[:p.Length]...)
+	b := append(p.Data, make([]byte, 9)...)
```

Comment:
Hm… if this triggers a reallocation, then we end up duplicating everything twice.

But I’m not sure if we want to have a conditional in here to avoid that in the case that should not happen.

Reply:
A reallocation should not happen since we allocated the needed 9 extra bytes before

Reply:
Exported function though. :wink: 

But yeah, I don’t think it’s probably worth it, because like you said, it’s use is almost exclusively internal, and we allocate enough space.

Reply:
Added some documentation about the required capacity